engine-tests/src/tests/eth_connector.rs

```diff
@@ -660,6 +660,193 @@ fn test_deposit_with_0x_prefix() {
+        to_yocto("100"), // initial balance
+    );
+
+    let recipient_address = Address::from_array([10u8; 20]);
+    let recipient_address_encoded = recipient_address.encode();
+    let message = [user_account_id, ":", "0x", &recipient_address_encoded].concat();
+    let proof = generate_dummy_proof(message, deposit_amount, 1);
+    let res = master_account.call(
+        contract.account_id(),
+        "deposit",
+        &proof.try_to_vec().unwrap(),
+        DEFAULT_GAS,
+        0,
+    );
+    let promises = res.promise_results();
+    res.assert_success();
+
+    let promise = &promises[promises.len() - 5];
+    assert_execution_status_failure(
+        promise.as_ref().unwrap().outcome().clone().status,
+        format!(
+            r#"CompilationError(CodeDoesNotExist {{ account_id: AccountId("{}") }}"#,
+            user_account_id
+        )
+        .as_str(),
+        "Expected failure in `ft_on_transfer` call, but deposit succeeded",
+    );
+
+    let user_account_balance =
+        get_eth_on_near_balance(&master_account, user_account_id, CONTRACT_ACC);
+    assert_eq!(user_account_balance, 0);
+    let aurora_balance = get_eth_on_near_balance(&master_account, CONTRACT_ACC, CONTRACT_ACC);
+    assert_eq!(aurora_balance, deposit_amount);
+    let address_balance = get_eth_balance(&master_account, recipient_address, CONTRACT_ACC);
+    assert_eq!(address_balance, 0);
+}
+
+#[test]
+fn test_deposit_eth_with_custom_connector_account() {
+    // In this test we make an ETH deposit using the message format for targeting
+    // an Aurora address, but use a different NEAR account than the Aurora Engine.
+    // Additionally, the target account implements `ft_on_transfer` so that it can
+    // receive the ETH perform some action with it. This is safe because the ETH is
+    // minted in the Engine first, then transferred to the target account using
+    // `ft_transfer_call`.
+    let (master_account, contract) = init(CUSTODIAN_ADDRESS);
+
+    let deposit_amount = 17;
+
+    let user_account_id = "some_user.root";
+    let _user_account = master_account.deploy(
+        &dummy_ft_receiver_bytes(),
+        user_account_id.parse().unwrap(),
+        to_yocto("100"), // initial balance
+    );
+
+    let recipient_address = Address::from_array([10u8; 20]);
+    let recipient_address_encoded = recipient_address.encode();
+    let message = [user_account_id, ":", "0x", &recipient_address_encoded].concat();
+    let proof = generate_dummy_proof(message, deposit_amount, 1);
+    let res = master_account.call(
+        contract.account_id(),
+        "deposit",
+        &proof.try_to_vec().unwrap(),
+        DEFAULT_GAS,
+        0,
+    );
+    let promises = res.promise_results();
+    for p in promises.iter() {
+        assert!(p.is_some());
+        let p = p.as_ref().unwrap();
+        if p.executor_id().as_str() == user_account_id {
+            // The ft_on_transfer implementation in the user's account generates this log.
```

Comment:
```suggestion
            // The `ft_on_transfer` implementation in the user's account generates this log.
```

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/598/commits/56d71542154fee9a25ec595d12d3c1ec1ee8146a